articles/how-to-collect-sensor-data-on-your-iphone-or-android-device-using-matlab/index.md

```diff
@@ -0,0 +1,111 @@
+```Matlab
+%% load data acquired by position data
+%load position data
+addpath('/MATLAB Drive/MobileSensorData')
+load sensorlog.mat
+```
+![loaded data](/how-to-collect-sensor-data-on-your-iphone-or-android-device-using-matlab/engineering-education/sensor_ten.png)
+
+We then introduce the haversine formula. This formula uses the latitudinal and longitudinal positions to calculate the distance. The latitudinal and longitudinal positions are in degrees. It means we will need to convert them into radians before using them. Also, in this formula, the radius of the earth is considered.
+```Matlab
+%have haversine formula to calculate distance
+latPos = deg2rad(Position.latitude);
+lonPos = deg2rad(Position.longitude);
+radiusEarth = 6370 ; % in Km
+```
+use vectorization to calculate total distance at every time step
+```matlab
+ii = 1:length(latPos)-1;
+```
+Since the data is large, it is first pre-processed. Once this is done, we then calculate the distance.
+```matlab
+% Prepare vector for calculation
+lat1 = latPos(ii);
+lat2 = latPos(ii+1);
+
+lon1 = lonPos(ii);
+lon2 = lonPos(ii+1);
+
+difflat = lat2-lat1;
+difflon = lon2 -lon1;
+
+aa = sin(difflat/2).^2 + cos(lat1).*cos(lat2).*sinc(difflon/2).^2;
+cc = 2.*atan2(sqrt(aa), sqrt(1-aa));
+dd = radiusEarth*cc;
+
+%% Total distance
+distMiles = sum(dd)*0.621;
+disp(distMiles)
+```
+When we execute the whole program, we get the distance traveled in kilometers. When we run the same script in the mobile Matlab, we get a similar answer as that of PC Matlab:
+
+![Output of the distance](/how-to-collect-sensor-data-on-your-iphone-or-android-device-using-matlab/engineering-education/sensor_nine.png)
+
+It is proof that these sensors are working and accurate. You can similarly use the remaining sensors to collect your data.
+
+### Conclusion
+Using a phone's sensor to collect data is easy. It involves the activation of the sensor and then beginning to the collection of data. It makes it an efficient way of collecting data in the absence of sensors. They provide accurate results which are reliable for any experiments. They are also easy to access since you only need to have the mobile Matlab app, and you are ready to collect the data. The significant advantage is that the data can be collected using a readily available resource(phone's sensor). You can be sure that if 80% of the day, you have your iPhone or Android phone with you, and you can collect any data anywhere and anytime.
```

Comment:
```suggestion
Using a phone's sensor to collect data is easy. It involves the activation of the sensor and then beginning to the collection of data. It makes it an efficient way of collecting data in the absence of sensors. They provide accurate results which are reliable for any experiments. 

They are also easy to access since you only need to have the mobile Matlab app, and you are ready to begin collection. 

The significant advantage is that the data can be collected using a readily available resource(phone's sensor). You can be sure that if 80% of the day, you have your iPhone or Android phone with you, and you can collect any data anywhere and anytime.
```